source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -6,8 +6,18 @@ namespace Clusters {
 namespace Redis {
 
 bool ClusterSlot::operator==(const Envoy::Extensions::Clusters::Redis::ClusterSlot& rhs) const {
-  return start_ == rhs.start_ && end_ == rhs.end_ && primary_ == rhs.primary_ &&
-         replicas_ == rhs.replicas_;
+  if (start_ != rhs.start_ || end_ != rhs.end_ || *primary_ != *rhs.primary_ ||
+      replicas_.size() != rhs.replicas_.size()) {
+    return false;
+  } else {
+    for (auto it1 = replicas_.begin(), it2 = rhs.replicas_.begin(); it1 != replicas_.end();
+         it1++, it2++) {
+      if (**it1 != **it2) {
+        return false;
+      }
+    }
```

Comment:
nit: this loop shouldn't be nested in `else` to lessen cognitive load.

Reply:
Thank you for your suggestion, I have add a new commit.